src/pycolmap/sensor/bitmap.cc

```diff
@@ -0,0 +1,126 @@
+#include "colmap/sensor/bitmap.h"
+
+#include "pycolmap/helpers.h"
+#include "pycolmap/pybind11_extension.h"
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+
+using namespace colmap;
+using namespace pybind11::literals;
+namespace py = pybind11;
+
+void BindBitmap(pybind11::module& m) {
+  py::class_<Bitmap>(m, "Bitmap")
+      .def(py::init<>())
+      .def("to_array",
+           [](const Bitmap& self) {
+             std::vector<ssize_t> shape = {static_cast<ssize_t>(self.Height()),
+                                           static_cast<ssize_t>(self.Width())};
+             const auto channels = static_cast<ssize_t>(self.Channels());
+             const bool is_rgb = self.IsRGB();
+             if (channels != 1) {
+               if (channels != 3) {
+                 throw std::runtime_error(
+                     "Can only convert grayscale or 3-channel RGB image to "
+                     "array");
+               }
+               shape.push_back(channels);
+             }
+             py::array_t<uint8_t> output(shape);
+             py::buffer_info output_into = output.request();
+             uint8_t* output_row_ptr =
+                 reinterpret_cast<uint8_t*>(output.request().ptr);
+             const size_t output_pitch =
+                 (output_into.shape.size() == 2)
+                     ? output_into.shape[1]
+                     : (output_into.shape[1] * output_into.shape[2]);
+             for (ssize_t y = 0; y < output_into.shape[0]; ++y) {
+               if (is_rgb) {
+                 for (ssize_t x = 0; x < output_into.shape[1]; ++x) {
+                   BitmapColor<uint8_t> color;
+                   THROW_CHECK(self.GetPixel(x, y, &color));
+                   output_row_ptr[3 * x] = color.r;
+                   output_row_ptr[3 * x + 1] = color.g;
+                   output_row_ptr[3 * x + 2] = color.b;
+                 }
```

Comment:
It would be nice to have a comment that explains why we cannot copy the row in this case. Is it about the ordering of the axes?